bin/population/pycbc_population_rates

```diff
@@ -0,0 +1,237 @@
+
+prob['lnm'] = rf.prob_lnm
+prob['imf'] = rf.prob_imf
+label['lnm'] = 'Flat in log'
+label['imf'] = 'Power-law'
+
+# Estimate the rates and make supporting plots
+vt, vol_time, sig_vt, inj_falloff = {}, {}, {}, {}
+for dist in distrs:
+    vt[dist] = si.estimate_vt(injections, mchirp_sampler[dist], prob[dist], thr_var = opts.thr_var, thr_val = opts.thr_val, min_mass = opts.min_mass, max_mass = opts.max_mass, max_mtotal = opts.max_mtotal)
+    
+    vol_time[dist], sig_vt[dist] = si.get_summed_vt(vt[dist])
+    inj_falloff[dist] = si.get_accumulated_falloff(vt[dist])
+
+#Include the calibration uncertainity
+vt_err_w_cal_uncrt = {}
+for dist in distrs:
+
+    vt_err_w_cal_uncrt[dist] = np.sqrt((vol_time[dist]*3*opts.cal_err/100.)**2 + sig_vt[dist]**2)
+    
+#for dist in distrs:
+#    hist4, bins= np.histogram(snr_falloff[dist], bins = 30, density=True) 
+#    width, center = (bins[1] - bins[0]), (bins[:-1] + bins[1:]) / 2
+#    plt.bar(center, hist4, align='center', width=width, alpha = 0.8, label = label[dist])
+#    plt.legend(loc = 'best')
+#plt.title('SNR Fall Off') 
+
+
+#Sabe background data and coincidences
+print opts.statmap_folder, opts.bank_folder, path, opts.thr_val, opts.min_tmplt_mass
+rf.save_bkg_falloff(opts.statmap_folder, opts.bank_folder, path, 8.0, 5.0)
+
+#Load background data and coincidences/ make some plots
+coincs = np.loadtxt(path + "/coincs.txt")
+bg_l, bg_h, bg_counts = np.loadtxt(path + "/background_bins.txt", unpack=True)
+bg_bins = np.append(bg_l, bg_h[-1])
+
+fg_stats = np.concatenate([inj_falloff[dist] for dist in distrs])
+fg_bins = np.logspace(np.log10(opts.thr_val), np.log10(np.max(fg_stats)), 101)
+fg_stats = fg_stats[fg_stats > opts.thr_val]
+
+ts = np.linspace(opts.thr_val, 11.0, 1000)
+plt.figure()
+plt.plot(ts, np.exp(rf.log_rho_bg(ts, bg_bins, bg_counts)), label=r'$p_0$')
+plt.plot(ts, np.exp(rf.log_rho_fg_mc(ts, fg_stats, fg_bins)), label=r'$p_1$')
+plt.yscale('log')
+plt.xlabel(r"$x'$")
+plt.ylabel(r"$p(x')$")
+plt.legend(loc='lower left')
+plt.axis(ymin=1e-3)
+plt.tight_layout()
+plt.savefig(path+'/fg-bg.png')
+
+#Load prior samples and fit a skew-log-normal to it
+with h5py.File(opts.prior_samples, "r") as f:
+    Rfl = np.array(f['flat/Rf'])
+    Rpl = np.array(f['power-law/Rf'])
+    
+alpha, mu, sigma = {}, {}, {}
+alpha['lnm'], mu['lnm'], sigma['lnm'] = rf.fit(Rfl)
+alpha['imf'], mu['imf'], sigma['imf'] = rf.fit(Rpl)
+
+#Estimate rates
+rate_data = {}
+rate_samples = {}
+
+log_fg_ratios = rf.log_rho_fg_mc(coincs, fg_stats, fg_bins) - rf.log_rho_bg(coincs, bg_bins, bg_counts)
+for dist in distrs:
+    rate_data[dist] = {'Ncoincs': len(log_fg_ratios),
+           'log_fg_ratios': log_fg_ratios,
+           'mu_vt': vol_time[dist]/1e9,
+           'sigma_vt': sig_vt[dist]/1e9,
+           'alpha': alpha[dist],
+           'mu': mu[dist],
+           'sigma': vt_err_w_cal_uncrt[dist]}
+    rate_fit = pystan.stan(file=opts.stan_file, data=rate_data[dist], iter=20000)
```

Comment:
FYI @cdcapano : this is directly from the reviewed O1 FGMC rates analysis written by Will Farr.  A .stan file is the specification of the statistical model that is being sampled, it would be order(50-100) lines so very impractical to include as such in the help message.  The call to pystan does the MC sampling of the model.  To continue doing what the O1 rates analysis did, it's necessary to continue using pystan *or* to replace it with some other statistical modelling and sampling package - but to review that (as yet non-existent) alternative - which would have to be coded up by somebody using another sampling module - would require showing it did the same as pystan, so in either case we need pystan.  Let's get the O1 analysis as-is into the pipeline first and worry about cleanliness of imports later.

Reply:
I am not using pystan anymore, but have introduced a function fg_mc in the file rates_functions.py, please have a look